TestLoadConfigFromProfile flakiness

## Failure

```
    --- FAIL: TestLoadConfigFromProfile/teleportHome_is_not_specified (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x100 pc=0x1042ffea0]

goroutine 58309 [running]:
testing.tRunner.func1.2({0x105a25300, 0x109872660})
	/Users/jnyckowski/go/go1.19.4/src/testing/testing.go:1396 +0x1c8
testing.tRunner.func1()
	/Users/jnyckowski/go/go1.19.4/src/testing/testing.go:1399 +0x378
panic({0x105a25300, 0x109872660})
	/Users/jnyckowski/go/go1.19.4/src/runtime/panic.go:884 +0x204
github.com/gravitational/teleport/tool/tctl/common.LoadConfigFromProfile(0x14007010e40, 0x14006c64000)
	/Users/jnyckowski/projects/teleport/tool/tctl/common/tctl.go:409 +0x650
github.com/gravitational/teleport/tool/tsh.TestLoadConfigFromProfile.func2(0x140065be420?)
	/Users/jnyckowski/projects/teleport/tool/tsh/tctl_test.go:92 +0x34
testing.tRunner(0x140067b8d00, 0x14006c74ff0)
	/Users/jnyckowski/go/go1.19.4/src/testing/testing.go:1446 +0x10c
created by testing.(*T).Run
	/Users/jnyckowski/go/go1.19.4/src/testing/testing.go:1493 +0x300

```
I've seen this failure when running this test locally on MacOS.

--- a/tool/tsh/tctl_test.go
+++ b/tool/tsh/tctl_test.go
@@ -80,11 +80,6 @@ func TestLoadConfigFromProfile(t *testing.T) {
 				TeleportHome: "some/dir/that/does/not/exist",
 			},
 			want: trace.NotFound("profile is not found"),
-		}, {
-			name: "teleportHome is not specified",
-			ccf:  &common.GlobalCLIFlags{},
-			cfg:  &servicecfg.Config{},
-			want: trace.NotFound("profile is not found"),
 		},
 	}
 	for _, tc := range tests {

